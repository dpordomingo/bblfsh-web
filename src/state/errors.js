const initialState = [];

export const ADD = 'bblfsh/errors/ADD';
export const SET = 'bblfsh/errors/SET';
export const REMOVE = 'bblfsh/errors/REMOVE';
export const CLEAR = 'bblfsh/errors/CLEAR';

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ADD:
      return state.concat(action.errors || []);
    case SET:
      return action.errors || [];
    case REMOVE:
      return [...state.slice(0, action.idx), ...state.slice(action.idx + 1)];
    case CLEAR:
      return [];
    default:
      return state;
  }
};

export const add = errors => ({
  type: ADD,
  errors,
});

export const set = errors => ({
  type: SET,
  errors,
});

export const remove = idx => ({
  type: REMOVE,
  idx,
});

export const clear = () => ({
  type: CLEAR,
});
