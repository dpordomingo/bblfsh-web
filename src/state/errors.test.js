import { reducer, initialState, ADD, SET, REMOVE, CLEAR } from './errors';

describe('errors/reducer', () => {
  it('ADD', () => {
    expect(
      reducer(initialState, { type: ADD, errors: ['error1'] })
    ).toMatchSnapshot();
    expect(
      reducer(initialState.concat(['error1']), {
        type: ADD,
        errors: ['error2'],
      })
    ).toMatchSnapshot();
  });

  it('SET', () => {
    expect(
      reducer(initialState.concat(['error1']), {
        type: SET,
        errors: ['error2'],
      })
    ).toMatchSnapshot();
  });

  it('REMOVE', () => {
    expect(
      reducer(initialState.concat(['error1', 'error2', 'error3']), {
        type: REMOVE,
        idx: 1,
      })
    ).toMatchSnapshot();
  });

  it('CLEAR', () => {
    expect(
      reducer(initialState.concat(['error1']), { type: CLEAR })
    ).toMatchSnapshot();
  });
});
