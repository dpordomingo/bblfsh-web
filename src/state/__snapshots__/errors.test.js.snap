// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`errors/reducer ADD 1`] = `
Array [
  "error1",
]
`;

exports[`errors/reducer ADD 2`] = `
Array [
  "error1",
  "error2",
]
`;

exports[`errors/reducer CLEAR 1`] = `Array []`;

exports[`errors/reducer REMOVE 1`] = `
Array [
  "error1",
  "error3",
]
`;

exports[`errors/reducer SET 1`] = `
Array [
  "error2",
]
`;
